test/js/ez-navigation-hub.js

```diff
@@ -7,4 +12,128 @@ describe('ez-navigation-hub', function() {
+            );
+        }
+        describe('highlight zone', function () {
+            it('should highlight zone contents', function () {
+                const zone = element.querySelector('[data-zone-identifier="contents"]');
+                const link = zone.querySelector('a');
+                const event = simulateClick(link);
+
+                assertEventHandled(event);
+                assert.isOk(zone.classList.contains(element.activeZoneClass));
+            });
+
+            it('should highlight zone other and unhighlight zone contents', function () {
+                const contentsZone = element.querySelector('[data-zone-identifier="contents"]');
+                const otherZone = element.querySelector('[data-zone-identifier="other"]');
+                const link = otherZone.querySelector('a');
+
+                element.activeZone = 'contents';
+                const event = simulateClick(link);
+                assertEventHandled(event);
+
+                assert.isNotOk(contentsZone.classList.contains(element.activeZoneClass), 'zone contents should not be highlighted');
+                assert.isOk(otherZone.classList.contains(element.activeZoneClass), 'zone other should be highlighted');
+            });
+        });
+
+        describe('update navigation', function () {
+            it('should display zone contents links', function () {
+                const link = element.querySelector('[data-zone-identifier="contents"] a');
+                const event = simulateClick(link);
+
+                assertEventHandled(event);
+                assert.isNotOk(element.querySelector('[data-zone="other"]').classList.contains('is-in-active-zone'), 'zone other links should not be displayed');
+                assert.isOk(element.querySelector('[data-zone="contents"]').classList.contains('is-in-active-zone'));
+
+            });
+
+            it('should display zone other links', function () {
+                const link = element.querySelector('[data-zone-identifier="other"] a');
+                const event = simulateClick(link);
+
+                assertEventHandled(event);
+                assert.isNotOk(element.querySelector('[data-zone="contents"]').classList.contains('is-in-active-zone'), 'zone content links should not be displayed');
+                assert.isOk(element.querySelector('[data-zone="other"]').classList.contains('is-in-active-zone'));
+            });
+        });
+    });
+
+    describe('match link url', function () {
+        it('Should highlight link', function () {
+            element.matchedLinkUrl = 'url1';
+            assert.isOk(element.querySelector('[href="url1"]').parentNode.classList.contains(element.matchedLinkClass), 'link1 should be highlighted');
+            element.matchedLinkUrl = 'url2';
+            assert.isNotOk(element.querySelector('[href="url1"]').parentNode.classList.contains(element.matchedLinkClass), 'link1 should not be highlighted anymore');
```

Comment:
please split this in 2 tests. One about highlighting the link, one about switching